pynucastro/rates/rate.py

```diff
@@ -240,9 +244,14 @@ def _set_print_representation(self):
                     rhs_other.append("nubar")
 
         for n, r in enumerate(treactants):
-            self.string += f"{r.c()}"
-            self.rid += f"{r}"
-            self.pretty_string += fr"{r.pretty}"
+            factor = ""
+            if self.stoichiometry:
+                c = self.stoichiometry[r]
+                if c != 1:
+                    factor = f"{c} "
```

Comment:
I think there is a potential issue when dealing with rates that have duplicate reactants or products, triple-alpha for example. Suppose you set `stoichiometry = {Nucleus("he"): 4, Nucleus("c12"): 1}`, then it shows up as `4 He4 + 4 He4 + 4 He4 --> C12`, I think we want it to be `4 He4 --> C12`.

Reply:
yeah, this is a good point.  Maybe we should always contract things down, so 3-alpha will normally appear as
3 He4 --> C12.  I think that's the cleanest way to do it.

Reply:
I think this is implemented.  I probably need to update some notebooks.